src/foam/u2/crunch/CapabilityStore.js

```diff
@@ -473,7 +475,19 @@ foam.CLASS({
             if ( ucj && ( ucj.status == this.CapabilityJunctionStatus.GRANTED
               || ucj.status == this.CapabilityJunctionStatus.PENDING ) ) return;
 
-            this.openWizard(cap, false);
+            let x = this.ctrl.__subContext__;
+            await this.grantAll(x, cap.id, this.subject);
+
+            var data = null;
+            if ( cap.id.endsWith("general-admission") )
+              data = net.nanopay.crunch.onboardingModels.SubmitData.create({ submitted: true });
```

Comment:
seems wrong with this hard coded part, looking for a better way.

Reply:
@KernelDeimos Thanks for the helpful comments!
I tried it and updated in another way as there are some cap.of data like SubmitData, InitialBusinessData which have submitted boolean property. Then if those caps. do not meet the requirements to get Granted, submitted needs to be back to false. Please review again.